### ISSUE TITLE
Question about generating pointcloud

### ISSUE THREAD

**Issue by Ridhwanluthra** [princeton-vl/RAFT-Stereo#26]
While generating pointcloud from the disparity generated using raft-stereo I end up getting something like thi:
![raft_ptc](https://user-images.githubusercontent.com/16349604/151890445-de144a3c-9051-4760-815e-4ab51ae9ad5e.png)
) 
Where there is a trail of points from the object to the background. The image I have posted is a top view.
I am using the standard equations to generate the pointcloud, am I missing something?

**Comment by lahavlipson** [comment]
I observed this as well, especially near object boundaries which end up getting smoothed by the network. 

The results look a lot cleaner if you filter the points using forward-backward consistency. The [bilinear_sampler](https://github.com/princeton-vl/RAFT-Stereo/blob/5c13878b617177da139cfeba79ac15b39b351963/core/utils/utils.py#L59) function should make it easy to sample a tensor using the predicted coordinates

**Comment by Ridhwanluthra** [comment]
Oh interesting, thank you so much, I'll look into it. Additionally, do you think their could be a way to get the network to not do this?

**Comment by lahavlipson** [comment]
Any network that uses an endpoint error loss for training will probably smooth over some object boundaries, so I'm not sure.

Also the point cloud may look better (but sparser) if you remove points whose gradient is larger than some threshold.

**Comment by Mythili-kannan** [comment]
I am also facing same issue(while generating Point clouds getting trailing points), can you please explain little more on
 how bilinear_sampler can be used to filter the trailing points and also 
 how to calculate gradient of points in pointcloud to remove, it will be very helpful.

**Comment by lahavlipson** [comment]
You can use forward-backward consistency to filter the points. The bilinear_sampler() function in this repo is just a wrapper for [torch.nn.functional.grid_sample](https://pytorch.org/docs/stable/generated/torch.nn.functional.grid_sample.html#torch-nn-functional-grid-sample), which will enable you to sample the backward flowfield using a forward one. You can produce a "backward" flowfield by just reversing the images and rerunning RAFT-Stereo.

A simpler method is to just exclude points whose predicted depth is very different from that of its neighbors.

**Comment by lahavlipson** [comment]
In case anyone comes across this, here is a Google Colab with code to produce a point cloud from the RAFT-Stereo predictions: https://colab.research.google.com/drive/1G8WJCQt9y55qxQH6QV6PpPvWEbd393g2?usp=sharing

**Comment by niubiplus2** [comment]
Hello, how to get the baseline of the cameras, please?
